x/group/msgs.go

```diff
@@ -239,6 +226,114 @@ func (m *MsgUpdateGroupMembers) GetGroupID() uint64 {
 	return m.GroupId
 }
 
+var _ sdk.Msg = &MsgCreateGroupWithPolicy{}
+var _ types.UnpackInterfacesMessage = MsgCreateGroupWithPolicy{}
+
+// NewMsgCreateGroupWithPolicy creates a new MsgCreateGroupWithPolicy.
+func NewMsgCreateGroupWithPolicy(admin string, members []Member, group_metadata []byte, group_policy_metadata []byte, groupPolicyAsAdmin bool, decisionPolicy DecisionPolicy) (*MsgCreateGroupWithPolicy, error) {
+	m := &MsgCreateGroupWithPolicy{
+		Admin:               admin,
+		Members:             members,
+		GroupMetadata:       group_metadata,
+		GroupPolicyMetadata: group_policy_metadata,
+		GroupPolicyAsAdmin:  groupPolicyAsAdmin,
+	}
+	err := m.SetDecisionPolicy(decisionPolicy)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+func (m *MsgCreateGroupWithPolicy) GetAdmin() string {
+	return m.Admin
+}
+
+func (m *MsgCreateGroupWithPolicy) GetGroupMetadata() []byte {
+	return m.GroupMetadata
+}
+
+func (m *MsgCreateGroupWithPolicy) GetGroupPolicyMetadata() []byte {
+	return m.GroupPolicyMetadata
+}
+
+func (m *MsgCreateGroupWithPolicy) GetGroupPolicyAsAdmin() bool {
+	return m.GroupPolicyAsAdmin
+}
```

Comment:
why do we need all these getters?

Reply:
getters for admin, groupMetadata and groupPolicyMetadata are not needed. `GroupPolicyAsAdmin` getter is used in `msg_server.go`

Reply:
you could just use `m.GroupPolicyAsAdmin` instead directly